### ISSUE TITLE
Talkback is not announcing the associated group header information

### ISSUE THREAD

**Issue by adelKhatib** [microsoft/fluentui#21827]
### Environment Information

- **Package version(s)**: 8.17.2
 Platform: Android
OS Version: Android 12
Device: Pixel 3Xl
Microsoft Teams Version: 1416/1.0.02021194504/1209 (Google Play Store Build)
Screen reader: Talkback

### Describe the issue:

While navigating over the dropdown items, when focus of Talkback lands over the first item of each group it is not announcing the associated label of the group header.

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/adelkhatib/pen/NWwMdbq

#### Actual behavior:

Talkback announce the focused item label without the group header associated with the item.

#### Expected behavior:
Talkback should announce the group header associated with the item.

**Note**: the issue repro on chrome with Narrator screen reader.

### Documentation describing expected behavior
**User Impact**: Label makes navigation easier, knowing the label of contents helps user to identify the category of items.
